### PR TITLE
Update dependencies + use kcd-scripts' husky

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,10 +17,11 @@
     "end-to-end",
     "e2e"
   ],
-  "author": "Kent C. Dodds <me@kentcdodds.com> (https://kentcdodds.com)",
+  "author": "Kent C. Dodds <kentcdodds@gmail.com> (https://kentcdodds.com)",
```
Comment (kentcdodds) [inline package.json]: Let's keep this as me@kentcdodds.com

Comment (MichaelDeBoey) [reply package.json]: I just copied it over from `generator-kcd-oss`, will change it again 🙂

File: package.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
   "author": "Kent C. Dodds <me@kentcdodds.com> (https://kentcdodds.com)",
   "license": "MIT",
   "engines": {
-    "node": ">=10"
+    "node": ">=10",
+    "npm": ">=6"
```
Comment (eps1lon) [inline package.json]: This engines are for consuming code. I don't think this library requires npm.

Comment (MichaelDeBoey) [reply package.json]: It's also checked I think

Comment (eps1lon) [reply package.json]: What do you mean by "it's checked"?

Comment (MichaelDeBoey) [reply package.json]: When you run `npm install`, the version is checked.
Just like you have when using `yarn`.

Comment (eps1lon) [reply package.json]: Yeah but it shouldn't. I can install this library just as well with yarn or pnpm.

Comment (MichaelDeBoey) [reply package.json]: The version is only checked when using `npm`

Comment (eps1lon) [reply package.json]: > You can also use the “engines” field to specify which versions of npm are capable of properly installing your program.

-- https://docs.npmjs.com/files/package.json#engines

Why do we need npm 6 to install `@testing-library/dom`?

Comment (MichaelDeBoey) [reply package.json]: It's not necessary, but just setting it as a sensible baseline
Most people are already using `npm@6`

Comment (eps1lon) [reply package.json]: Any code that is not necessary should be removed.

Comment (kentcdodds) [reply package.json]: Yeah, I think we should probably just remove the npm engine. It's really not necessary.

Comment (MichaelDeBoey) [reply package.json]: I'll remove it then 🙂

**Pull request merged**